### ISSUE TITLE
Flaky build: jaeger_test.go:171 runtime error: index out of range

### ISSUE THREAD

**Issue by jacobgreenleaf** [uber/zanzibar#506]
e.g. https://travis-ci.org/uber/zanzibar/jobs/447999869

```
--- FAIL: TestHTTPEndpointToHTTPClientWithUpstreamSpan (0.20s)
	Error Trace:	jaeger_test.go:171
	Error:      	Not equal: 
	            	expected: 2
	            	actual: 0
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range
goroutine 7 [running]:
testing.tRunner.func1(0xc4203403c0)
	/home/travis/.gimme/versions/go1.10.4.linux.amd64/src/testing/testing.go:742 +0x29d
panic(0xb74020, 0x112fae0)
	/home/travis/.gimme/versions/go1.10.4.linux.amd64/src/runtime/panic.go:502 +0x229
github.com/uber/zanzibar/test.TestHTTPEndpointToHTTPClientWithUpstreamSpan(0xc4203403c0)
	/home/travis/gopath/src/github.com/uber/zanzibar/test/jaeger_test.go:172 +0xfb0
testing.tRunner(0xc4203403c0, 0xc856f0)
	/home/travis/.gimme/versions/go1.10.4.linux.amd64/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/home/travis/.gimme/versions/go1.10.4.linux.amd64/src/testing/testing.go:824 +0x2e0
FAIL	github.com/uber/zanzibar/test	0.414s
```

**Comment by l-w-2017** [comment]
This should be fixed already by #504

**Comment by l-w-2017** [comment]
Job is green now.
